analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -233,6 +234,7 @@ def __get_compiler_err(cmd):
         except OSError as oerr:
             LOG.error("Error during process execution: " + cmd + '\n' +
                       oerr.strerror + "\n")
+            return ''
```

Comment:
Do we really need this empty string? `None` as return value seems more Python style for me in case of an error. The `if` condition below can prevents its splitting too.

Reply:
Returning None could cause more problems if the value is not checked properly, almost every string  function would throw an exception on it.

Reply:
Personally I prefer error messages, otherwise bugs may remain hidden. But ok.

Reply:
My problem with it is that we return an empty string only if an `OSError` occures. What if this code throws an other error? We will return `None` in this case. It would be better to return with the same type in case of every errors.

This way I would prefer @bruntib suggestion to return None instead of an empty string.